pkg/util/pki/nameconstraints_test.go

```diff
@@ -49,7 +49,7 @@ func TestMarshalNameConstraints(t *testing.T) {
 			expectedResult: pkix.Extension{
 				Id:       OIDExtensionNameConstraints,
 				Critical: true,
-				Value:    []byte{0x30, 0x57, 0xa0, 0x3, 0x1, 0x1, 0xff, 0xa1, 0xf, 0x30, 0xd, 0x13, 0xb, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0xa3, 0x10, 0x30, 0xe, 0x30, 0xc, 0x4, 0x4, 0xc0, 0xa8, 0x0, 0x0, 0x4, 0x4, 0xff, 0xff, 0xff, 0x0, 0xa5, 0x14, 0x30, 0x12, 0xc, 0x10, 0x75, 0x73, 0x65, 0x72, 0x40, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0xa7, 0x17, 0x30, 0x15, 0x13, 0x13, 0x68, 0x74, 0x74, 0x70, 0x73, 0x3a, 0x2f, 0x2f, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d},
+				Value:    []byte{0x30, 0x48, 0xa0, 0x46, 0x30, 0xd, 0x82, 0xb, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0x30, 0xa, 0x87, 0x8, 0xc0, 0xa8, 0x0, 0x0, 0xff, 0xff, 0xff, 0x0, 0x30, 0x12, 0x81, 0x10, 0x75, 0x73, 0x65, 0x72, 0x40, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0x30, 0x15, 0x86, 0x13, 0x68, 0x74, 0x74, 0x70, 0x73, 0x3a, 0x2f, 0x2f, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d},
```

Comment:
@tanujd11 could you maybe add some testcases that compare the nameconstraints values with values generated using the openssl CLI tool? (place the DER in the test code + the openssl command to generate the CSR & extract the Nameconstraints extension to use it in this test).

Reply:
Hi @inteon I have added the TestMarshalNameConstraints output from a file which is created by openssl, The generation command is also added as comment. Thanks. PTAL